test/topology_custom/test_read_repair.py

```diff
@@ -212,7 +212,9 @@ async def test_incremental_read_repair(data_class, workdir, manager):
     seed = int(time.time())
     logger.info(f"random-seed: {seed}")
     random.seed(seed)
-    cmdline = ["--hinted-handoff-enabled", "0", "--query-tombstone-page-limit", "1000"]
+    cmdline = ["--hinted-handoff-enabled", "0",
+               "--query-tombstone-page-limit", "10",
```

Comment:
Nitpick: (feel free to ignore, because it's a pre-existing issue)
It's sad we rely more and more on each test starting a Scylla cluster from scratch. It's not really necessary, you can see in test/cql-pytest/test_tombstone_limit.py an example test that can use any Scylla cluster and temporarily reduce query_tombstone_page_limit to 10. That approach doesn't allow for parallel tests on the same cluster, but it at least allows reusing the cluster - which could shave the remaining 13.5 seconds of this test to much less.

Reply:
Yes, I suspect almost half of the test's run-time is startup and teardown. In theory clusters *are* reused, although I don't know under what conditions.

Reply:
Does adding custom command-line parameters make a cluster dirty? @kostja, @kbr-scylla ?

Here, I can use the config table and revert my config changes at the end of the test, if that will make this cluster reusable.

Reply:
Adding a node makes the cluster dirty.

Since test/topology_custom starts with 0-size cluster, it means every test has to add at least one node, means there is no cluster reuse in this suite.

Reply:
I know this was discussed many times.... sorry for piling on more. I think maybe instead of different suite for different defaults (that are impossible to guess from the suite name), we should merge these suites as much as possible, and have tests communicate what kind of cluster they need, via fixtures. We could have fixtures for the most common setups, as well as a fixture for a "custom" one. Tests depending on this fixture, will get 0-node cluster.
We can also use decorators if fixtures prove problematic.

Reply:
Agreed.

But merging suites should also come with an improvement to test.py and cluster pool so we better utilize available resources on test machine. Currently it's very stupid, each suite has its constant `pool_size` which determines how many concurrent clusters will run within this suite, and different suites have no concurrency control w.r.t. each other (they all run at the same time IIUC).

*And after everyone agreed with each other, we all went home and nothing was done.*